Exploring Data and Small Observable Trends

In [70]:
import pandas as pd

In [71]:
df = pd.read_csv("2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv")

In [72]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   order_id        5000 non-null   int64 
 1   shop_id         5000 non-null   int64 
 2   user_id         5000 non-null   int64 
 3   order_amount    5000 non-null   int64 
 4   total_items     5000 non-null   int64 
 5   payment_method  5000 non-null   object
 6   created_at      5000 non-null   object
dtypes: int64(5), object(2)
memory usage: 273.6+ KB


In [73]:
df.head(10)

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   
5         6       58      882           138            1    credit_card   
6         7       87      915           149            1           cash   
7         8       22      761           292            2           cash   
8         9       64      914           266            2          debit   
9        10       52      788           146            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11  
5  2017-03-14 15:25:01  
6  2017-03-01 21:37:57  
7   2017-03-08 2:05:38  
8  2017-03-17 20:56:50  
9  2017-03-30 21:08:26

In [74]:
df.isnull().sum()

order_id          0
shop_id           0
user_id           0
order_amount      0
total_items       0
payment_method    0
created_at        0
dtype: int64

In [75]:
df.describe()

order_id      shop_id      user_id   order_amount  total_items
count  5000.000000  5000.000000  5000.000000    5000.000000   5000.00000
mean   2500.500000    50.078800   849.092400    3145.128000      8.78720
std    1443.520003    29.006118    87.798982   41282.539349    116.32032
min       1.000000     1.000000   607.000000      90.000000      1.00000
25%    1250.750000    24.000000   775.000000     163.000000      1.00000
50%    2500.500000    50.000000   849.000000     284.000000      2.00000
75%    3750.250000    75.000000   925.000000     390.000000      3.00000
max    5000.000000   100.000000   999.000000  704000.000000   2000.00000

In [76]:
#import matplotlib.pyplot as plt
#%matplotlib inline
#import seaborn as sns
#plt.scatter(df['created_at'], df["order_amount"])
#plt.show()

In [77]:
df[df['order_amount'] == df['order_amount'].max()]

order_id  shop_id  user_id  order_amount  total_items payment_method  \
15          16       42      607        704000         2000    credit_card   
60          61       42      607        704000         2000    credit_card   
520        521       42      607        704000         2000    credit_card   
1104      1105       42      607        704000         2000    credit_card   
1362      1363       42      607        704000         2000    credit_card   
1436      1437       42      607        704000         2000    credit_card   
1562      1563       42      607        704000         2000    credit_card   
1602      1603       42      607        704000         2000    credit_card   
2153      2154       42      607        704000         2000    credit_card   
2297      2298       42      607        704000         2000    credit_card   
2835      2836       42      607        704000         2000    credit_card   
2969      2970       42      607        704000         2000    credit_card   
3332      3333       42      607        704000         2000    credit_card   
4056      4057       42      607        704000         2000    credit_card   
4646      4647       42      607        704000         2000    credit_card   
4868      4869       42      607        704000         2000    credit_card   
4882      4883       42      607        704000         2000    credit_card   

              created_at  
15    2017-03-07 4:00:00  
60    2017-03-04 4:00:00  
520   2017-03-02 4:00:00  
1104  2017-03-24 4:00:00  
1362  2017-03-15 4:00:00  
1436  2017-03-11 4:00:00  
1562  2017-03-19 4:00:00  
1602  2017-03-17 4:00:00  
2153  2017-03-12 4:00:00  
2297  2017-03-07 4:00:00  
2835  2017-03-28 4:00:00  
2969  2017-03-28 4:00:00  
3332  2017-03-24 4:00:00  
4056  2017-03-28 4:00:00  
4646  2017-03-02 4:00:00  
4868  2017-03-22 4:00:00  
4882  2017-03-25 4:00:00

In [78]:
df.groupby('shop_id').total_items.sum().sort_values()

shop_id
85       67
56       69
16       71
44       71
38       72
      ...  
53      130
71      130
84      132
13      136
42    34063
Name: total_items, Length: 100, dtype: int64

Question 1.


    A.

In [79]:
#Seeing Adjusted AOV Based on total orders divided by total items
total_revenue = df['order_amount'].sum()
total_items_ordered = df['total_items'].sum()

adjusted_AOV = total_revenue / total_items_ordered
adjusted_AOV

357.92152221412965

In [80]:
#Seeing Adjusted AOV per shopID
adjusted_AOV_grouped = df.groupby('shop_id').order_amount.sum() / df.groupby('shop_id').total_items.sum()
adjusted_AOV_grouped.sort_values()

shop_id
92        90.0
2         94.0
32       101.0
100      111.0
7        112.0
        ...   
99       195.0
89       196.0
12       201.0
42       352.0
78     25725.0
Length: 100, dtype: float64

In [81]:
#Seeing Adjusted AOV without shop 78
df_exclusive= adjusted_AOV_grouped.sort_values()[:-1]
df_exclusive.mean()

152.26262626262627

The AOV is now adjusted more appropriately per shop. The data is much less skewed because Shop 78's AOV is scaled much closer to the rest of the shops' AOV. Shop 78 is going to be an outlier. So we did the AOV without Shop 78, with the the AOV being 152.26. The AOV for Shop 78 is 25725

In [89]:
#Seeing the Median of the Adjusted Order Value
adjusted_AOV_grouped.median()



153.0

B.


Taking the median of the AOV is another way of seeing how accurate the Adjusted AOV is. This gauges is very close to the original AOV, so the median of the Adjusted AOV is an accurate way of telling if the Adjusted AOV is a good estimate.

C.
The value of this new way of determining the AOV is 153. This is extremely close compared to the Adjusted AOV

Question 2:

A:

SELECT ShipperName, COUNT(Orders.ShipperID) 
FROM Orders 
INNER JOIN Shippers 
ON Orders.ShipperID = Shippers.ShipperID 
GROUP BY Orders.ShipperID 
LIMIT 1;

Speedy Express --> 54 Orders


B:

SELECT LastName, COUNT(Orders.EmployeeID) 
FROM Orders 
INNER JOIN Employees 
ON Orders.EmployeeID = Employees.EmployeeID 
GROUP BY Orders.EmployeeID 
ORDER BY COUNT(Orders.EmployeeID) DESC 
LIMIT 1;

Peacock --> 40 orders


C:

SELECT ProductName, SUM(Quantity) as Total FROM Orders AS o
INNER JOIN OrderDetails AS d ON d.OrderID = o.OrderID
INNER JOIN Products AS p ON p.ProductID = d.ProductID
INNER JOIN Customers AS c ON c.customerID = o.CustomerID 
WHERE Country = "Germany"
GROUP BY p.ProductID
ORDER BY Total DESC
LIMIT 1;


Boston Crab Meat --> 160 orders